test/apis/user-from-existing-patient.ts

```diff
@@ -0,0 +1,159 @@
+import "isomorphic-fetch";
+import {medTechApi, MedTechApi} from "../../src/apis/medTechApi";
+import {User} from "../../src/models/User";
+import {Patient} from "../../src/models/Patient";
+import {webcrypto} from "crypto";
+import {hex2ua} from "@icure/api";
+import {getEnvVariables, setLocalStorage, TestUtils} from "../test-utils";
+import {Address} from "../../src/models/Address";
+import {Telecom} from "../../src/models/Telecom";
+import {assert} from "chai";
+import {ICureRegistrationEmail} from "../../src/utils/gatewayMessageFactory";
+import {HealthcareProfessional} from "../../src/models/HealthcareProfessional";
+
+setLocalStorage(fetch);
+
+const {iCureUrl: iCureUrl, hcpUserName: hcpUserName, hcpPassword: hcpPassword, hcpPrivKey: hcpPrivKey,
+      msgGtwUrl: msgGtwUrl} = getEnvVariables()
+
+let medtechApi: MedTechApi;
+let hcpUser: User;
+let hcp: HealthcareProfessional
+
+describe("A Healthcare Party", () => {
+
+  before(async () => {
+    medtechApi = await medTechApi()
+      .withICureBasePath(iCureUrl)
+      .withUserName(hcpUserName)
+      .withPassword(hcpPassword)
+      .withAuthServerUrl(msgGtwUrl)
+      .withCrypto(webcrypto as any)
+      .build();
+
+    hcpUser = await medtechApi.userApi.getLoggedUser();
+    await medtechApi.cryptoApi.loadKeyPairsAsTextInBrowserLocalStorage(
+      hcpUser.healthcarePartyId ?? hcpUser.patientId ?? hcpUser.deviceId!,
+      hex2ua(hcpPrivKey)
+    );
+
+    const hcpEmail = await TestUtils.getTempEmail();
+    hcp = await medtechApi.healthcareProfessionalApi.getHealthcareProfessional(hcpUser.healthcarePartyId!);
+    hcp.addresses = [new Address({
+      addressType: "home",
+      description: "London",
+      telecoms: [
+        new Telecom({
+          telecomType: "email",
+          telecomNumber: hcpEmail
+        })
+      ]
+    })]
+  });
+
+  async function userFromPatient(patient: Patient) {
+    const existingPatient = await medtechApi.patientApi.createOrModifyPatient(patient);
+    assert(!!existingPatient);
+
+    const messageFactory = new ICureRegistrationEmail(
+      hcp,
+      "test",
+      "iCure",
+      existingPatient
+    )
+    await medtechApi.userApi.createAndInviteUser(existingPatient, messageFactory);
+  }
+
+  it("should be able to create a new User from an existing Patient", async () => {
+    const email = await TestUtils.getTempEmail();
+    const newPatient = new Patient({
+      firstName: "Marc",
+      lastName: "Specter",
+      addresses: [new Address({
+        addressType: "home",
+        description: "London",
+        telecoms: [
+          new Telecom({
+            telecomType: "email",
+            telecomNumber: email
+          })
+        ]
+      })]
+    });
+    await userFromPatient(newPatient);
```

Comment:
Add some asssertions to be sure the user is properly created

Reply:
Added in the userFromPatient method, as it is shared by several tests